### PR TITLE
Add bucket resolver interface and use it for put,update,remove and ge…

### PR DESCRIPTION
**Pull request by geirst**
…t messages in DocumentApiConverter.

@vekterli and @toregge please review

### CODE REVIEW

**Review by toregge — APPROVED** [review]
LGTM

File: storage/src/tests/storageserver/documentapiconvertertest.cpp [hunk 1]
```diff
@@ -120,15 +141,46 @@ void DocumentApiConverterTest::testForwardedPut()
...
+    auto updateReply = dynamic_cast<api::UpdateReply*>(storageReply.get());
+    CPPUNIT_ASSERT(updateReply);
+
+    auto mbusMsg = _converter->toDocumentAPI(*updateCmd, _repo);
+    auto mbusUpdate = dynamic_cast<documentapi::UpdateDocumentMessage*>(mbusMsg.get());
+    CPPUNIT_ASSERT(mbusUpdate);
+    CPPUNIT_ASSERT((&mbusUpdate->getDocumentUpdate()) == update.get());
+    CPPUNIT_ASSERT_EQUAL(api::Timestamp(1234), mbusUpdate->getOldTimestamp());
+    CPPUNIT_ASSERT_EQUAL(api::Timestamp(5678), mbusUpdate->getNewTimestamp());
+}
```
Comment (vekterli) [inline storage/src/tests/storageserver/documentapiconvertertest.cpp]: I feel that this test ideally could be broken up into smaller, distinct tests. Currently is a bit hard to follow since it tests all update-related message conversions basically at the same time. Consider refactoring later, not a blocker for this PR.

**Review by vekterli — APPROVED** [review]
:+1:
